### PR TITLE
[native] migrate use-cases of makePromiseContract to structured bindings

### PR DESCRIPTION
**Pull request by yfeldblum**
Summary: It is desirable to have a contract struct type with names other than `first` and `second`, which is all we get with `std::pair`. To facilitate the switch, migrate to structured bindings.

Differential Revision: D55564951

### CODE REVIEW

File: presto-native-execution/presto_cpp/main/http/tests/HttpTest.cpp [hunk 1]
```diff
@@ -336,15 +336,15 @@ TEST_P(HttpTestSuite, httpConnectTimeout) {
...
+  requestState1->requestPromise = std::move(promise1);
+  auto [promise2, future2] = folly::makePromiseContract<bool>();
+  requestState2->requestPromise = std::move(promise2);
 
-  std::thread thread1([&]() {
+  std::thread thread1([&, &future2 = future2]() {
     auto responseFuture = sendGet(client1.get(), "/async/msg1");
     try {
-      std::move(promisePair1.second).wait();
+      std::move(future2).wait();
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/http/tests/HttpTest.cpp:347]: future1?

File: presto-native-execution/presto_cpp/main/http/tests/HttpTest.cpp [hunk 2]
```diff
@@ -336,15 +336,15 @@ TEST_P(HttpTestSuite, httpConnectTimeout) {
...
-  requestState1->requestPromise = std::move(promisePair1.first);
-  auto promisePair2 = folly::makePromiseContract<bool>();
-  requestState2->requestPromise = std::move(promisePair2.first);
+  auto [promise1, future1] = folly::makePromiseContract<bool>();
+  requestState1->requestPromise = std::move(promise1);
+  auto [promise2, future2] = folly::makePromiseContract<bool>();
+  requestState2->requestPromise = std::move(promise2);
 
-  std::thread thread1([&]() {
+  std::thread thread1([&, &future2 = future2]() {
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/http/tests/HttpTest.cpp:344]:
```
&future = future1
```

**Review by xiaoxmeng — COMMENTED** [review]
@yfeldblum thanks!

File: presto-native-execution/presto_cpp/main/http/tests/HttpTest.cpp [hunk 3]
```diff
@@ -336,15 +336,15 @@ TEST_P(HttpTestSuite, httpConnectTimeout) {
...
-  requestState1->requestPromise = std::move(promisePair1.first);
-  auto promisePair2 = folly::makePromiseContract<bool>();
-  requestState2->requestPromise = std::move(promisePair2.first);
+  auto [promise1, future1] = folly::makePromiseContract<bool>();
+  requestState1->requestPromise = std::move(promise1);
+  auto [promise2, future2] = folly::makePromiseContract<bool>();
+  requestState2->requestPromise = std::move(promise2);
 
-  std::thread thread1([&]() {
+  std::thread thread1([&, &future1 = future1]() {
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/http/tests/HttpTest.cpp:344]:
```
nit: s/future1/future/
```

**Review by xiaoxmeng — DISMISSED** [review]

**Comment by yfeldblum** [comment]
Replaced by: https://github.com/prestodb/presto/pull/22574.

**Pull request closed without merging**